src/index.d.ts

```diff
@@ -51,29 +51,51 @@ export interface CommandReturn<A, P> {
   _P: P
 }
 
-export type CommandArgsNoInvs<A extends IOTSParams, R> = {
+export type CommandArgsNoInvsNoDeps<A extends IOTSParams, R> = {
   params: A,
   run: CommandRun<IOTSDictToType<A>, R>,
-  invalidates?: never
+  invalidates?: never,
+  dependencies?: never
 };
 
 export type Invalidates = { [k: string]: QueryReturn<any, any> };
 
 type InvA<I extends Invalidates> = BROKEN_FlattenObject<{ [k in keyof I]: I[k]['_A'] }>;
 
-export type CommandArgs<A extends IOTSParams, I extends Invalidates, R> = (
-  ObjectOverwrite<CommandArgsNoInvs<A, R>, {
+export type CommandArgsNoDeps<A extends IOTSParams, I extends Invalidates, R> = (
+  ObjectOverwrite<CommandArgsNoInvsNoDeps<A, R>, {
     run: CommandRun<IOTSDictToType<A> & {[k in keyof I]: I[k]['_A']}, R>,
     invalidates: I
   }>
 );
 
-export function Command<A extends IOTSParams, R>(args: CommandArgsNoInvs<A, R>): CommandReturn<IOTSDictToType<A>, R>;
+export type CommandArgsNoInvs<A extends IOTSParams, D extends Dependencies, R> = (
+  ObjectOverwrite<CommandArgsNoInvsNoDeps<A, R>, {
+    run: CommandRun<IOTSDictToType<A> & {[k in keyof D]: D[k]['_P']}, R>,
+    dependencies: D
+  }>
+);
+
+export type CommandArgs<A extends IOTSParams, I extends Invalidates, D extends Dependencies, R> = (
+  ObjectOverwrite<CommandArgsNoInvs<A, D, R>, {
+    run: CommandRun<IOTSDictToType<A> & {[k in keyof I]: I[k]['_A']} & {[k in keyof D]: D[k]['_P']}, R>,
+    invalidates: I
+  }>
+);
+
+export function Command<A extends IOTSParams, R>(args: CommandArgsNoInvsNoDeps<A, R>): CommandReturn<IOTSDictToType<A>, R>;
+
+export function Command<A extends IOTSParams, R, I extends Invalidates>(
+  args: CommandArgsNoDeps<A, I, R>
+): CommandReturn<IOTSDictToType<A> & InvA<I>, R>;
 
-export function Command<A extends IOTSParams, R, I1 extends Invalidates>(
-  args: CommandArgs<A, I1, R>
-): CommandReturn<IOTSDictToType<A> & InvA<I1>, R>;
+export function Command<A extends IOTSParams, R, D extends Dependencies>(
+  args: CommandArgsNoInvs<A, D, R>
+): CommandReturn<IOTSDictToType<A> & DepA<D> & {[k in keyof D]?: D[k]['_P']}, R>;
 
+export function Command<A extends IOTSParams, R, I extends Invalidates, D extends Dependencies>(
+  args: CommandArgs<A, I, D, R>
+): CommandReturn<IOTSDictToType<A> & InvA<I> & DepA<D>  & {[k in keyof D]?: D[k]['_P']}, R>;
```

Comment:
A few things:
- we could define a `DepP<D>` helper since this is repeated a few times
- I'm thinking that, if `DepP<D>` is passed, than `DepA<D>` is not really required anymore.. but building the type correctly (either a dependency params A or its resolved value P) becomes quite complex I believe..